### PR TITLE
update AddonExample

### PR DESCRIPTION
**Pull request by qitana**
Update AddonExample Project.

I referred to the cactbot code.
Thanks to @quisquous.

### CODE REVIEW

File: AddonExample/packages.config [hunk 1]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Newtonsoft.Json" version="12.0.3" targetFramework="net45" />
+</packages>
```
Comment (ngld) [inline AddonExample/packages.config]: Can you please migrate this file to `<PackageReference>`s? If you right-click this file in Visual Studio, there should be an option to do that.

File: AddonExample/AddonExampleEventSource.cs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+        public void LogInfo(string format, params object[] args)
+        {
+            this.Log(LogLevel.Warning, format, args);
+        }
+
+        public void LogWarning(string format, params object[] args)
+        {
+            this.Log(LogLevel.Info, format, args);
+        }
```
Comment (ngld) [inline AddonExample/AddonExampleEventSource.cs]: Why did you add the `Log*` methods?
Calling `this.LogDebug("Some stuff with {0} value.", "a");` doesn't seem any simpler than `this.Log(LogLevel.Debug, "Some stuff with {0} value.", "a");`.

File: AddonExample/AddonExampleEventSource.cs [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+            Config.SaveConfig(config);
+        }
+
+        public override void Start()
+        {
+            // Reset Event Dispatcher
+            OnAddonExampleOriginalTimerFired -= (e) => DispatchToJS(e);
+            OnAddonExampleOriginalTimerFired += (e) => DispatchToJS(e);
+            OnAddonExampleEmbeddedTimerFired -= (e) => DispatchToJS(e);
+            OnAddonExampleEmbeddedTimerFired += (e) => DispatchToJS(e);
```
Comment (ngld) [inline AddonExample/AddonExampleEventSource.cs]: Why do you remove and add the event handlers here? Does that change anything?

File: AddonExample/AddonExampleEventSource.cs [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+            // Start Original Timer
+            originalTimer = new System.Timers.Timer()
+            {
+                Interval = 1000,
+                AutoReset = true,
+            };
+            originalTimer.Elapsed += (obj, args) =>
+            {
+                OnAddonExampleOriginalTimerFired(new JSEvents.OriginalTimerFiredEvent(Config.ExampleString + " fired!"));
```
Comment (ngld) [inline AddonExample/AddonExampleEventSource.cs]: Wouldn't it be much simpler to do this here?
```csharp
DispatchEvent(JObject.FromObject(new {
  type = "onAddonExampleOriginalTimerFiredEvent",
  message = Config.ExampleString + " fired!"
}));
```
Then you can remove the `OnAddonExampleOriginalTimerFired` event, the `OriginalTimerFiredEvent` class, the `DispatchToJS` method and you'll have less overhead.

File: AddonExample/AddonExample.csproj [hunk 5]
```diff
@@ -9,8 +9,9 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>AddonExample</RootNamespace>
     <AssemblyName>AddonExample</AssemblyName>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.7.1</TargetFrameworkVersion>
```
Comment (ngld) [inline AddonExample/AddonExample.csproj]: Can you please change this to `4.6.1`? I just changed this since apparently a lot of users have an older .NET installation and we don't really use anything that requires `4.7.1`.

File: AddonExample/AddonExample.cs [hunk 6]
```diff
@@ -1,57 +1,56 @@
...
         {
-            return new AddonExampleOverlayConfig(name);
+            get { return "Addon Example"; }
         }
 
-        public System.Windows.Forms.Control CreateOverlayConfigControlInstance(IOverlay overlay)
+        public string Description
         {
-            return new AddonExampleOverlayConfigPanel();
+            get { return "Addon Example Description"; }
```
Comment (ngld) [inline AddonExample/AddonExample.cs]: You can remove the `Name` and `Description` properties. OverlayPlugin doesn't use them.

**Review by ngld — CHANGES_REQUESTED** [review]
I'm sorry it took me so long to review this.
Thanks a lot!!

I've added some notes and questions. Since this will serve as a starting point to others, I want to keep it as simple as possible which is why I've added some questions about the `DispatchToJS`, `JSEvents` and `Log*` parts.
They look like you copied them from Cactbot's code. IMO they aren't necessary here and only make the example more complicated than it needs to be.

File: AddonExample/AddonExample.csproj (hunk 5 above)
Comment (qitana) [reply AddonExample/AddonExample.csproj]: Changed by 296d77a

File: AddonExample/AddonExample.cs (hunk 6 above)
Comment (qitana) [reply AddonExample/AddonExample.cs]: Delete them by d7924cc

File: AddonExample/packages.config (hunk 1 above)
Comment (qitana) [reply AddonExample/packages.config]: Changed to PackageReference by 9a8e11d

File: AddonExample/AddonExampleEventSource.cs (hunk 3 above)
Comment (qitana) [reply AddonExample/AddonExampleEventSource.cs]: I noticed that there was no need to use Event/Delegate.
Deleted them by 2c281aa

File: AddonExample/AddonExampleEventSource.cs (hunk 4 above)
Comment (qitana) [reply AddonExample/AddonExampleEventSource.cs]: fixed this by 2c281aa

**Comment by qitana** [comment]
Thanks to review.
I fixed problems which was pointed out.

**Comment by ngld** [comment]
Thanks a lot!
